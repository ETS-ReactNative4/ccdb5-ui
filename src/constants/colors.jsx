/* eslint-disable no-inline-comments, camelcase */
module.exports = Object.freeze( {
  BriteCharts: {
    regular: '#20aa3f',
    medium: '#ADDC91',
    light: '#C7E5B3'
  },
  DataLens: {
    0: '#2cb34a',
    1: '#addc91',
    2: '#257675',
    3: '#9ec4c3',
    4: '#0072ce',
    5: '#96c4ed',
    6: '#254b87',
    7: '#9daecc',
    8: '#b4267a',
    9: '#dc9cbf',
    10: '#a2a3a4'
  }
} )
