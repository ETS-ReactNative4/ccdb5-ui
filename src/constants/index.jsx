// Internal triggers
export const PAGE_CHANGED              = 'PAGE_CHANGED'
export const SEARCH_CHANGED            = 'SEARCH_CHANGED'
export const SIZE_CHANGED              = 'SIZE_CHANGED'
export const SORT_CHANGED              = 'SORT_CHANGED'
export const URL_CHANGED               = 'URL_CHANGED'
export const FILTER_CHANGED            = 'FILTER_CHANGED'
export const FILTER_REMOVED            = 'FILTER_REMOVED'
export const FILTER_ALL_REMOVED        = 'FILTER_ALL_REMOVED'

// External Triggers
export const COMPLAINTS_RECEIVED       = 'COMPLAINTS_RECEIVED'
export const COMPLAINTS_FAILED         = 'COMPLAINTS_FAILED'

// Useful constants
export const knownFilters = [
  'company',
  'company_public_response',
  'company_response',
  'consumer_consent_provided',
  'consumer_disputed',
  'issue',
  'product',
  'state',
  'submitted_via',
  'tag',
  'timely',
  'zip_code'
]

export const SLUG_SEPARATOR            = 'â€¢'
