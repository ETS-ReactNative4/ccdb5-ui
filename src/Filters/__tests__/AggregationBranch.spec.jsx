import React from 'react'
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux'
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import renderer from 'react-test-renderer';
import ReduxAggregationBranch, { AggregationBranch } from '../AggregationBranch'
import { SLUG_SEPARATOR } from '../../constants'

// ----------------------------------------------------------------------------
// Setup 

const item = {
  key: 'foo',
  doc_count: 99
}

const subitems = [
  { key: 'bar', doc_count: 90 },
  { key: 'baz', doc_count: 5 },
  { key: 'qaz', doc_count: 4 },
]

function setupEnzyme() {
  const props = {
    item: item,
    subitems: subitems,
    fieldName: "issue"
  }

  const target = shallow(<AggregationBranch {...props} />);

  return {
    props,
    target
  }
}

function setupSnapshot() {
  const middlewares = [thunk]
  const mockStore = configureMockStore(middlewares)
  const store = mockStore({
    query: {
      issue: ['foo' + SLUG_SEPARATOR + 'bar']
    }
  })

  return renderer.create(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ReduxAggregationBranch item={item} 
                                subitems={subitems}
                                fieldName="issue" />
      </IntlProvider>
    </Provider>
  )
}

// ----------------------------------------------------------------------------
// Test 

describe('component::AggregationBranch', () => {
  describe('initial state', () => {
    it('renders without crashing', () => {
      const target = setupSnapshot()
      const tree = target.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('toggle behavior', () => {
    it('shows the children when the label is clicked', () => {
      const { target, props } = setupEnzyme()
      const theButton = target.find('.toggle button')

      expect(target.state('showChildren')).toEqual(false);
      theButton.simulate('click');
      expect(target.state('showChildren')).toEqual(true);
    })  
  })
})
