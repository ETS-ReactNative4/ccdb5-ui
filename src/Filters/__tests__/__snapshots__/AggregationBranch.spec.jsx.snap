exports[`component::AggregationBranch initial state renders without crashing 1`] = `
<div
  className="aggregation-branch">
  <li
    className="flex-fixed layout-row parent">
    <input
      aria-label="foo"
      checked={false}
      className="flex-fixed"
      onClick={undefined}
      type="checkbox" />
    <div
      className="flex-all toggle">
      <button
        className="a-btn a-btn__link hover"
        onClick={[Function]}>
        <span>
          foo
        </span>
        <span
          className="cf-icon cf-icon-up" />
      </button>
    </div>
    <span
      className="flex-fixed parent-count">
      <span>
        99
      </span>
    </span>
  </li>
  <ul
    className="children">
    <li
      className="flex-fixed layout-row">
      <input
        aria-label="foo•bar"
        checked={true}
        className="flex-fixed"
        onClick={[Function]}
        type="checkbox" />
      <span
        className="flex-all bucket-key">
        bar
      </span>
      <span
        className="flex-fixed bucket-count">
        <span>
          90
        </span>
      </span>
    </li>
    <li
      className="flex-fixed layout-row">
      <input
        aria-label="foo•baz"
        checked={false}
        className="flex-fixed"
        onClick={[Function]}
        type="checkbox" />
      <span
        className="flex-all bucket-key">
        baz
      </span>
      <span
        className="flex-fixed bucket-count">
        <span>
          5
        </span>
      </span>
    </li>
    <li
      className="flex-fixed layout-row">
      <input
        aria-label="foo•qaz"
        checked={false}
        className="flex-fixed"
        onClick={[Function]}
        type="checkbox" />
      <span
        className="flex-all bucket-key">
        qaz
      </span>
      <span
        className="flex-fixed bucket-count">
        <span>
          4
        </span>
      </span>
    </li>
  </ul>
</div>
`;
